### ISSUE TITLE
Add parameter singleFile

### ISSUE THREAD

**Issue by GordeevArt** [asyncapi/html-template#61]
Motivation: publicate asyncapi into services that accept only single documents.

Add parameter **singleFile** to get html with scripts and styles in one file.

```
"singleFile": {
        "description": "If set this parameter to true generate single html file with scripts and styles inside",
        "required": false,
        "default": false
      }
```
Make something like this in index.html:

```
  {% if params.singleFile %}
    <style type="text/css">
    {% include "template/css/tailwind.min.css" %}
    </style>
    <style type="text/css">
    {% include "template/css/atom-one-dark.min.css" %}
    </style>
    <style type="text/css">
    {% include "template/css/main.css" %}
    </style>
  {% else %}
  <link href="css/tailwind.min.css" rel="stylesheet">
  <link rel="stylesheet" href="css/atom-one-dark.min.css" />
  <link rel="stylesheet" href="css/main.css" />
  {% endif %}
```
```
  {% if params.singleFile %}
    <script defer type="text/javascript">
    {% include "template/js/highlight.min.js" %}

    {% include "template/js/main.js" %}
    </script>
  {% else %}
  <script src="js/highlight.min.js" type="application/javascript"></script>
  <script src="js/main.js" type="application/javascript"></script>
  {% endif %}
```

**Comment by derberg** [comment]
@GordeevArt looks good to me, I like it, I think that one day someone asked about it. Did you think about creating a PR for it. The only missing part I see is to add a hook that would react to this parameter and would remove all unnecessary files from the output directory, something similar to https://github.com/asyncapi/java-spring-template/blob/master/hooks/02_removeNotRelevantParts.js. You can access params in `hooks -> const params = generator.templateParams;` and then something like `params.singleFile === 'true'`

**Comment by GordeevArt** [comment]
@derberg there is a problem with --watch-template when removing the directories by the hooks. Infinite recreating.

**Comment by derberg** [comment]
@GordeevArt crap, could you please report a bug [here](https://github.com/asyncapi/generator/issues)?

**Comment by GordeevArt** [comment]
@derberg yep. What about unpassed checks? https://github.com/asyncapi/html-template/pull/62/checks

How can it be fixed?

**Comment by derberg** [comment]
@GordeevArt our pipelines are based on GitHub Actions and when we added them, they did not support nicely `fork` workflows. Welcome message could not be posted by the bot as on the PR from fork it has only `read` GitHub Token. They changed it now and introduced an alternative pull request event that takes GitHub workflows files from the upstream and not the fork (which makes it secure to assign `read/write` token. We just didn’t have time to update our workflows in all our repositories. I hope I explained it well.

In theory, it is enough to replace [this](https://github.com/asyncapi/html-template/blob/master/.github/workflows/welcome-first-time-contrib.yml#L4) with `pull_request_target` but I didn't have time yet to test it
